specs/_features/eipxxxx/beacon-chain.md

```diff
@@ -0,0 +1,627 @@
+    """
+    output: Set[ValidatorIndex] = set()
+    committee_indices = get_committee_indices(attestation.committee_bits)
+    committee_offset = 0
+    for index in committee_indices:
+        committee = get_beacon_committee(state, attestation.data.slot, index)
+        committee_attesters = set(
+            index for i, index in enumerate(committee) if attestation.aggregation_bits[committee_offset + i])
+        output = output.union(committee_attesters)
+
+        committee_offset += len(committee)
+
+    ptc = get_ptc(state, attestation.data.slot)
+    return [i for i in output if i not in ptc]
+```
+
+#### `get_payload_attesting_indices`
+
+```python
+def get_payload_attesting_indices(state: BeaconState, slot: Slot, 
+                                  payload_attestation: PayloadAttestation) -> Set[ValidatorIndex]:
+    """
+    Return the set of attesting indices corresponding to ``payload_attestation``.
+    """
+    ptc = get_ptc(state, slot)
+    return set(index for i, index in enumerate(ptc) if payload_attestation.aggregation_bits[i])
+```
+
+
+#### `get_indexed_payload_attestation`
+
+```python
+def get_indexed_payload_attestation(state: BeaconState, slot: Slot, 
+                                    payload_attestation: PayloadAttestation) -> IndexedPayloadAttestation:
+    """
+    Return the indexed payload attestation corresponding to ``payload_attestation``.
+    """
+    attesting_indices = get_payload_attesting_indices(state, slot, payload_attestation)
+
+    return IndexedPayloadAttestation(
+        attesting_indices=sorted(attesting_indices),
+        data=payload_attestation.data,
+        signature=payload_attestation.signature,
+    )
+```
+
+## Beacon chain state transition function
+
+*Note*: state transition is fundamentally modified in EIP-XXXX. The full state transition is broken in two parts, first importing a signed block and then importing an execution payload.
+
+The post-state corresponding to a pre-state `state` and a signed beacon block `signed_block` is defined as `state_transition(state, signed_block)`. State transitions that trigger an unhandled exception (e.g. a failed `assert` or an out-of-range list access) are considered invalid. State transitions that cause a `uint64` overflow or underflow are also considered invalid. 
+
+The post-state corresponding to a pre-state `state` and a signed execution payload envelope `signed_envelope` is defined as `process_execution_payload(state, signed_envelope)`. State transitions that trigger an unhandled exception (e.g. a failed `assert` or an out-of-range list access) are considered invalid. State transitions that cause a `uint64` overflow or underflow are also considered invalid. 
+
+### Block processing
+
+```python
+def process_block(state: BeaconState, block: BeaconBlock) -> None:
+    process_block_header(state, block)
+    process_withdrawals(state) [Modified in EIP-XXXX]
```

Comment:
We can instead remove withdrawals from here, process them with the execution payload and then there is no need to track the last withdrawals root in the Beacon state.